### PR TITLE
[Components with more than 1 service] LPS-179567 Refactor ScreenNavigationCategory components from Commerce modules part 4

### PR DESCRIPTION
**Pull request by daiazius**

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:startup

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CommerceDiscountDetailsScreenNavigationEntry.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	private static final Log _log = LogFactoryUtil.getLog(
+		CommerceDiscountDetailsScreenNavigationEntry.class);
+
+	@Reference
+	private CommerceCurrencyLocalService _commerceCurrencyLocalService;
+
+	@Reference(
+		target = "(component.name=com.liferay.commerce.pricing.web.internal.frontend.taglib.servlet.taglib.CommerceDiscountDetailsScreenNavigationCategory)"
+	)
+	private ScreenNavigationCategory
```
Comment (tinatian) [inline modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CommerceDiscountDetailsScreenNavigationEntry.java:134]: @daiazius @rafaprax no need to add reference to category, use the constants directly. This is to make it consistent with current ScreenNavigationEntry, also when rendering on page, you will get category first, then find its entries, no need to have this kind of dependecy.

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by daiazius** [comment]
ci:test:startup

**Comment by daiazius** [comment]
Hey, I just changed the commit messages, no changes in the code.

**Review by rafaprax — APPROVED** [review]

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CPInstancePriceListsScreenNavigationCategory.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 	@Override
 	public String getCategoryKey() {
-		return "price-lists";
+		return CPInstanceScreenNavigationConstants.CATEGORY_KEY_PRICE_LISTS;
```
Comment (rafaprax) [inline modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CPInstancePriceListsScreenNavigationCategory.java:40]: Hey @daiazius, rename it to CPInstanceScreenNavigationConstants.CATEGORY_KEY_CP_INSTANCE_PRICE_LISTS, please. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by Kvale1733** [comment]
@rafaprax No regressions. Only a semver issue.
```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-product-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.product.servlet.taglib.ui.constants MINOR      1.1.0      1.1.0      1.2.0      VERSION INCREASE REQUIRED
     [exec]     	<   class      com.liferay.commerce.product.servlet.taglib.ui.constants.CPInstanceScreenNavigationConstants
     [exec]     		+   field      CATEGORY_KEY_PRICE_LISTS
     [exec]     			+   access     final
     [exec]     			+   access     static
     [exec]     			+   constant   price-lists
     [exec]     			+   return     java.lang.String
     [exec]     [Baseline Warning] Bundle Version Change Recommended: 62.1.0
     [exec]         at org.junit.Assert.fail(Assert.java:88)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-03/test-1-40/test-portal-acceptance-pullrequest(master)/981/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Review by rafaprax — APPROVED** [review]

**Comment by daiazius** [comment]
Hello, I only pushed a baseline commit, no changes in logic.

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by daiazius** [comment]
ci:test:startup

**Review by rafaprax — APPROVED** [review]

**Comment by zhangyan0701** [comment]
No regressions found.

**Pull request closed without merging**